### PR TITLE
chore: hide private values in debug and serialize

### PR DESCRIPTION
**Pull request by Cifko**
Description
---
Added `derivative` macro to get rid of the debug prints.

Motivation and Context
---
Hide all private values that can be exposed accidentally via debug or serializing.

How Has This Been Tested?
---
integration tests only

### CODE REVIEW

**Review by sdbondi — DISMISSED** [review]
Nice!

File: comms/src/peer_manager/identity_signature.rs [hunk 1]
```diff
@@ -38,9 +39,11 @@ use crate::{
 };
 
 /// Signature that secures the peer identity
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[derive(Derivative, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[derivative(Debug)]
 pub struct IdentitySignature {
     version: u8,
+    #[derivative(Debug = "ignore")]
```
Comment (sdbondi) [inline comms/src/peer_manager/identity_signature.rs]: Signature contains public values (SecretKey is a public scalar `s` in this case) so can remove this ignore

**Review by sdbondi — COMMENTED** [review]
LGTM, just the one signature that does not have to be ignored

File: comms/src/peer_manager/identity_signature.rs (hunk 1 above)
Comment (Cifko) [reply comms/src/peer_manager/identity_signature.rs]: Done.

**Comment by Cifko** [comment]
These ones are required to be serialized:

`base_layer\core\src\transactions\transaction_components\unblinded_output.rs`
  - UnblindedOutput
    - spending_key
    
`base_layer\core\src\transactions\transaction_protocol\sender.rs`
  - RawTransactionInfo
    - recipient_sender_offset_private_keys
    - private_commitment_nonces
    - gamma
    - private_nonce
    
`base_layer\key_manager\src\key_manager.rs`
  - KeyManager
    - seed
    - branch_seed
    
`base_layer\key_manager\src\wasm.rs
  - DerivedKeypair
    - private_key
   
`comms\src\peer_manager\node_identity.rs`
  - NodeIdentity
    - secret_key

**Review by sdbondi — APPROVED** [review]

**Pull request merged**